### PR TITLE
feat!: Add support for creating GitHub App with organizations

### PR DESCRIPTION
**Pull request by wenked**
Fixes: #3210.

**BREAKING CHANGE:** The `CreateApp` function now requires two arguments: `AppManifest` and `orgName`, to support creating apps with organizations.

The issue occurred because the create app URL path was only set up for creating personal GitHub apps.

My solution was to add an additional argument to the function to provide the necessary information for creating organizational GitHub Apps. The second argument, `AppConfig`, now includes configuration structs with an `OwnerType` that can be either personal or organizational, and you can provide the organization name by passing `OrgName` in the `AppConfig` struct.

For more details, see the issue: [https://github.com/google/go-github/issues/3210](https://github.com/google/go-github/issues/3210).

### CODE REVIEW

File: scrape/scrape.go [hunk 1]
```diff
@@ -128,6 +128,7 @@ func (c *Client) Authenticate(username, password, otpseed string) error {
 		values.Set("password", password)
 	}
 	resp, err := fetchAndSubmitForm(c.Client, "https://github.com/login", setPassword)
+
```
Comment (gmlewis) [inline scrape/scrape.go:131]: Please revert this change.  It is idiomatic Go to tightly couple the handling of an error with the statement that generates the error without a blank line in between.

File: scrape/apps.go [hunk 2]
```diff
@@ -133,9 +134,28 @@ type AppManifest struct {
...
+	OwnerType OwnerType
+	// OrgName is the name of the organization if the OwnerType is organizational.
+	OrgName string
+}
+
+func (c *Client) CreateApp(m *AppManifest, ac *AppConfig) (*http.Response, error) {
+	path := "/settings/apps/new"
+
+	if ac.OwnerType == "organizational" {
+		path = fmt.Sprintf("/organizations/%s/settings/apps/new", ac.OrgName)
```
Comment (gmlewis) [inline scrape/apps.go:155]: nit: for consistency in this repo when using `grep` or `ripgrep` to search for endpoint URLs:

```suggestion
		path = fmt.Sprintf("/organizations/%v/settings/apps/new", ac.OrgName)
```

File: scrape/apps.go [hunk 3]
```diff
@@ -133,9 +134,28 @@ type AppManifest struct {
 	DefaultPermissions *github.InstallationPermissions `json:"default_permissions,omitempty"`
 }
 
-// CreateApp creates a new GitHub App with the given manifest configuration.
-func (c *Client) CreateApp(m *AppManifest) (*http.Response, error) {
-	u, err := c.baseURL.Parse("/settings/apps/new")
+type OwnerType string
```
Comment (gmlewis) [inline scrape/apps.go:137]: Please add a `godoc`-style comment for this exported type.

File: scrape/apps.go [hunk 4]
```diff
@@ -133,9 +134,28 @@ type AppManifest struct {
...
+
+const (
+	OwnerTypePersonal     OwnerType = "personal"
+	OwnerTypeOrganization OwnerType = "organizational"
+)
+
+type AppConfig struct {
```
Comment (gmlewis) [inline scrape/apps.go:144]: Please add a `godoc`-style comment for this exported type.

File: scrape/apps.go [hunk 5]
```diff
@@ -133,9 +134,28 @@ type AppManifest struct {
...
+	// OwnerType specifies the type of GitHub account owner (personal or organizational).
+	OwnerType OwnerType
+	// OrgName is the name of the organization if the OwnerType is organizational.
+	OrgName string
+}
+
+func (c *Client) CreateApp(m *AppManifest, ac *AppConfig) (*http.Response, error) {
+	path := "/settings/apps/new"
```
Comment (gmlewis) [inline scrape/apps.go:152]: nit: for repo consistency:

```suggestion
	url := "/settings/apps/new"
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @wenked !
Please address the few minor comments.

@willnorris - since this involves `scrape`, do you have any comments or concerns regarding the breaking change in this PR?

File: scrape/apps.go [hunk 6]
```diff
@@ -133,9 +134,28 @@ type AppManifest struct {
...
+)
+
+type AppConfig struct {
+	// OwnerType specifies the type of GitHub account owner (personal or organizational).
+	OwnerType OwnerType
+	// OrgName is the name of the organization if the OwnerType is organizational.
+	OrgName string
+}
+
+func (c *Client) CreateApp(m *AppManifest, ac *AppConfig) (*http.Response, error) {
```
Comment (willnorris) [inline scrape/apps.go:151]: This AppConfig struct feels over-engineered.  In the case of creating a personal app, you can't provide a nil AppConfig, since line 154 will panic, but otherwise none of the values get used.  I *think* all you really want is the organization name, right?  So why not just pass an `orgName string` parameter, and if it's an empty string then create a personal app?

**Review by willnorris — COMMENTED** [review]
I don't have any sense on who is using this package (not many people, looking at pkg.go.dev), so I don't have strong feelings either way on the change.  looks fine to me.

File: scrape/apps.go (hunk 6 above)
Comment (wenked) [reply scrape/apps.go:151]: You're right, I'll do that instead .

**Review by gmlewis — APPROVED** [review]
Thank you, @wenked !
LGTM.
Merging.
